src/Machete/ValueExtensions.cs

```diff
@@ -174,5 +174,25 @@ public static T Or<T>(this Value<T> value, T other)
 
             return getter(source.Value) ?? Schema.Layout.Missing<T>();
         }
+
+        /// <summary>
+        /// Returns a count of segments in <see cref="ValueList{TValue}"/>
+        /// </summary>
+        /// <param name="source"></param>
+        /// <typeparam name="TValue"></typeparam>
+        /// <returns></returns>
+        public static int Count<TValue>(this ValueList<TValue> source)
```

Comment:
I can imagine a scenario where all I would like to do is determine how many items I am getting before I attempt to get them